iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnBuffers.cpp

```diff
@@ -1405,12 +1410,49 @@ static LogicalResult createAndPropagateBufferUsedForResultTensors(
     FuncOp funcOp, OutputBufferMap &outputBufferMap,
     TensorToBufferMap &resultTensorToBufferMap) {
   OpBuilder builder(funcOp.getBody());
-  auto walkResult = funcOp.walk(
-      [&](IREE::HAL::InterfaceStoreTensorOp storeTensorOp) -> WalkResult {
-        return createAndPropagateBufferUsedForResultTensor(
-            storeTensorOp, outputBufferMap, resultTensorToBufferMap, builder);
-      });
-  return failure(walkResult.wasInterrupted());
+  for (auto &block : funcOp.body().getBlocks()) {
```

Comment:
Maybe we can restrict this a bit and just check that the function has a single block. Without that we will need liveness analysis etc. to do a proper allocation.

Reply:
SG, done.